### PR TITLE
Added smart_proxy_reports new package

### PR DESCRIPTION
**Pull request by lzap**
For plugin updates, please indicate which repos this should be built into:

* Nightly

RPM version: https://github.com/theforeman/foreman-packaging/pull/7168

### CODE REVIEW

**Comment by lzap** [comment]
Do not merge, I am going to rebase this to 0.0.5.

**Comment by ekohl** [comment]
Converted it to a draft so that's clear.

**Comment by lzap** [comment]
Renamed "host_reports" to "reports", bumped version to 0.0.5. This is now ready for re-review, thanks guys.

**Comment by lzap** [comment]
Rebased to see what is wrong.

**Comment by lzap** [comment]
Could not find 'json' (>= 2.3.0) - did find: [json-2.1.0]

Hmmm I guess 2.1 is fine.

**Comment by evgeni** [comment]
```
 Could not find 'rack' (>= 2.1.0) - did find: [rack-2.0.6] (Gem::MissingSpecVersionError)
```

File: plugins/smart_proxy_reports/control [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Testsuite: autopkgtest-pkg-ruby
+XS-Ruby-Versions: all
+Rules-Requires-Root: no
+
+Package: ruby-smart-proxy-reports
+Architecture: all
+XB-Ruby-Versions: ${ruby:Versions}
+Depends: ruby | ruby-interpreter,
+         foreman-proxy (>= 2.5.0),
+         ruby-rack (>= 1.1.0),
```
Comment (evgeni) [inline plugins/smart_proxy_reports/control:19]: this is clearly wrong:

https://github.com/theforeman/smart_proxy_reports/blob/ce29f1b57b0090945e8c999debf9a58cd7390331/smart_proxy_reports.gemspec#L18-L19

File: plugins/smart_proxy_reports/control [hunk 2]
```diff
@@ -0,0 +1,23 @@
+Source: ruby-smart-proxy-reports
+Section: ruby
+Priority: optional
+Maintainer: Lukas Zapletal <lzap+pub@redhat.com>
+Build-Depends: debhelper (>= 11~),
+               ruby-rack (>= 1.1.0),
```
Comment (evgeni) [inline plugins/smart_proxy_reports/control:6]: this is clearly wrong:

https://github.com/theforeman/smart_proxy_reports/blob/ce29f1b57b0090945e8c999debf9a58cd7390331/smart_proxy_reports.gemspec#L18-L19

**Comment by ekohl** [comment]
Why does it even depend on rack? I'd expect the Smart Proxy itself to take care of that.

**Comment by evgeni** [comment]
same q for json actually, you don't need to depend on it (heck, you don't even require json anywhere)

**Comment by lzap** [comment]
Hmmm I see that json is now part of Ruby. That can be removed I guess.

Rake is used in tests, so dev only dependency. But I have it defined in the `Gemfile`, so what is preferred - to have it in the Gemfile or in the gemspec?

**Comment by lzap** [comment]
Removed the deps from gemspec, pushed out 0.0.7 - my lucky number!!! Notice that I created this PR for 0.0.2...

You will hardly find a worst packager than me.

**Comment by lzap** [comment]
Huston, we are green. T minus five.

File: plugins/smart_proxy_reports/copyright [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: smart_proxy_shellhooks
+Source: http://github.com/theforeman/smart_proxy_shellhooks
```
Comment (evgeni) [inline plugins/smart_proxy_reports/copyright:3]: I don't think so.

**Comment by evgeni** [comment]
tiny nit, and could you please adjust the commit message to match the new name of the gem? :)

**Comment by ekohl** [comment]
> Rake is used in tests, so dev only dependency. But I have it defined in the `Gemfile`, so what is preferred - to have it in the Gemfile or in the gemspec?

From a packaging perspective I don't think there's a difference since both are ignored.

File: plugins/smart_proxy_reports/copyright (hunk 3 above)
Comment (lzap) [reply plugins/smart_proxy_reports/copyright]: Yes, yes. I cheated. So what? :)

**Comment by lzap** [comment]
Cool, rebased.

**Review by evgeni — APPROVED** [review]

**Pull request merged**